### PR TITLE
restrict --chain flag setting for op-reth node & fix help info

### PR DESCRIPTION
**Pull request by roberto-bayardo**
- restrict set of allowed chains for op-reth to base and base-goerli only since other chains are untested and probably don't yet work, with base being the default
- fixes the command-line help documentation for the --chain flag for op-reth to print out the correct list of supported chains
- changes base-goerli chain name to snake case (base_goerli) to be consistent with previous behavior

### CODE REVIEW

File: bin/reth/src/args/utils.rs [hunk 1]
```diff
@@ -31,7 +38,7 @@ pub fn chain_spec_value_parser(s: &str) -> eyre::Result<Arc<ChainSpec>, eyre::Er
         "holesky" => HOLESKY.clone(),
         "dev" => DEV.clone(),
         #[cfg(feature = "optimism")]
-        "base-goerli" => BASE_GOERLI.clone(),
+        "base_goerli" => BASE_GOERLI.clone(),
```
Comment (mattsse) [inline bin/reth/src/args/utils.rs:41]: we can keep both variants here

File: bin/reth/src/args/utils.rs [hunk 2]
```diff
@@ -15,6 +15,13 @@ use std::{
 #[cfg(feature = "optimism")]
 use reth_primitives::{BASE_GOERLI, BASE_MAINNET};
 
+#[cfg(feature = "optimism")]
+/// Chains supported by op-reth. First value should be used as the default.
+pub const ALLOWED_CHAINS: &[&str] = &["base", "base_goerli"];
+#[cfg(not(feature = "optimism"))]
+/// Chains supported by reth. First value should be used as the default.
+pub const ALLOWED_CHAINS: &[&str] = &["mainnet", "sepolia", "goerli", "holesky", "dev"];
```
Comment (mattsse) [inline bin/reth/src/args/utils.rs:23]: I'd like to rename this to `SUPPORTED_CHAINS`.

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, I like this.

only naming nits

File: bin/reth/src/args/utils.rs (hunk 2 above)
Comment (roberto-bayardo) [reply bin/reth/src/args/utils.rs]: done

File: bin/reth/src/args/utils.rs (hunk 1 above)
Comment (roberto-bayardo) [reply bin/reth/src/args/utils.rs:46]: done

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**